packages/demo/usecase-demo/src/utils/localStorageContentMap.ts

```diff
@@ -1,37 +1,78 @@
+import { Address } from 'wagmi';
 import { LOCAL_STORAGE_PREFIX } from '@/utils/localStorage.ts';
 
+type TaskData = {
+  wallet_id: Address;
+  protected_data_address: string;
+  completed_task_id: string;
+};
+
+const STORED_DATA_KEY = `${LOCAL_STORAGE_PREFIX}_savedTaskId`;
+
 /**
  * localStorage cache for completed iExec tasks.
  * ie. Content is available on IPFS *but* the key needs not to have been re-generated in the meantime!
  */
 
 export function saveCompletedTaskId({
+  walletId,
   protectedDataAddress,
   completedTaskId,
 }: {
+  walletId: Address;
   protectedDataAddress: string;
   completedTaskId: string;
 }) {
-  localStorage.setItem(
-    `${LOCAL_STORAGE_PREFIX}_${protectedDataAddress}`,
-    completedTaskId
+  const storedData = localStorage.getItem(STORED_DATA_KEY);
+  const data: TaskData[] = storedData ? JSON.parse(storedData) : [];
```

Comment:
Je suis toujours un peu allergique aux variables nommées de manière trop générique: `data`.
Est-ce qu'on peut renommer ?

Simplement en `tasks` par exemple ?

```suggestion
  const tasks: TaskData[] = storedData ? JSON.parse(storedData) : [];
```